src/main/java/org/team1540/robot2024/subsystems/vision/AprilTagVision.java

```diff
@@ -0,0 +1,112 @@
+            Supplier<Double> elevatorVelocitySupplierMPS) {
+        this.frontCameraIO = frontCameraIO;
+        this.rearCameraIO = rearCameraIO;
+        this.visionPoseConsumer = visionPoseConsumer;
+        this.elevatorHeightSupplierMeters = elevatorHeightSupplierMeters;
+        this.elevatorVelocitySupplierMPS = elevatorVelocitySupplierMPS;
+    }
+
+    @Override
+    public void periodic() {
+        // TODO: need to change if one camera is stationary
+        frontCameraIO.setPoseOffset(
+                new Pose3d(
+                        FRONT_CAMERA_POSE.getX(),
+                        FRONT_CAMERA_POSE.getY(),
+                        FRONT_CAMERA_POSE.getZ() + elevatorHeightSupplierMeters.get(),
+                        FRONT_CAMERA_POSE.getRotation())
+        );
+        rearCameraIO.setPoseOffset(
+                new Pose3d(
+                        REAR_CAMERA_POSE.getX(),
+                        REAR_CAMERA_POSE.getY(),
+                        REAR_CAMERA_POSE.getZ() + elevatorHeightSupplierMeters.get(),
+                        REAR_CAMERA_POSE.getRotation())
+        );
+
+        frontCameraIO.updateInputs(frontCameraInputs);
+        rearCameraIO.updateInputs(rearCameraInputs);
+        Logger.processInputs("Vision/FrontCamera", frontCameraInputs);
+        Logger.processInputs("Vision/RearCamera", rearCameraInputs);
+
+        if (frontCameraInputs.lastMeasurementTimestampSecs > frontPose.timestampSecs()) {
+            frontPose = new TimestampedVisionPose(
+                    frontCameraInputs.lastMeasurementTimestampSecs,
+                    frontCameraInputs.estimatedPoseMeters,
+                    frontCameraInputs.seenTagIDs,
+                    frontCameraInputs.tagPosesMeters);
+        }
+        if (rearCameraInputs.lastMeasurementTimestampSecs > rearPose.timestampSecs()) {
+            rearPose = new TimestampedVisionPose(
+                    rearCameraInputs.lastMeasurementTimestampSecs,
+                    rearCameraInputs.estimatedPoseMeters,
+                    rearCameraInputs.seenTagIDs,
+                    rearCameraInputs.tagPosesMeters);
+        }
+
+        TimestampedVisionPose latestPose = getEstimatedPose();
+        Logger.recordOutput("Vision/EstimatedPose", latestPose == null ? new Pose2d() : latestPose.poseMeters());
+        visionPoseConsumer.accept(latestPose);
+    }
+
+    /**
+     * Gets the estimated pose by fusing individual computed poses from each camera.
+     * Returns null if no tags seen, in simulation, or if the elevator is moving
+     * too fast
+     */
+    public TimestampedVisionPose getEstimatedPose() {
+        if (Constants.currentMode == Constants.Mode.SIM) return null;
+        if (frontPose.getNumTagsSeen() < 1 && rearPose.getNumTagsSeen() < 1) return null;
+        if (Math.abs(elevatorVelocitySupplierMPS.get()) > MAX_ACCEPTED_ELEVATOR_SPEED_MPS) return null; // TODO: need to change if one camera is stationary
```

Comment:
shouldn't this be handled by VisionPoseAcceptor?

Reply:
This is in fact true, however doing it this way does make it easier to only get the pose from one camera if it is stationary while the other one is on the elevator. I can change how VisionPoseAcceptor is implemented so that I can do that in here if that's preferable.

Reply:
Alternatively, make VisionPoseAcceptor's features part of AprilTagVision

Reply:
Ok I saw this a bit too late, but VisionPoseAcceptor is now only used in AprilTagVision